tests/app/service/test_rest.py

```diff
@@ -590,6 +727,127 @@ def test_update_service(client, notify_db, sample_service):
     assert result['data']['organisation_type'] == 'other'
 
 
+def test_update_service_with_valid_provider(
+        admin_request, notify_db, sample_service, ses_provider, current_sms_provider
+):
+    data = {
+        'email_provider_id': str(ses_provider.id),
+        'sms_provider_id': str(current_sms_provider.id),
+    }
+
+    resp = admin_request.post(
+        'service.update_service',
+        service_id=sample_service.id,
+        _data=data,
+        _expected_status=200
+    )
+    assert resp['data']['email_provider_id'] == str(ses_provider.id)
+    assert resp['data']['sms_provider_id'] == str(current_sms_provider.id)
+
+    service_db = Service.query.get(resp['data']['id'])
+
+    assert service_db.email_provider_id == ses_provider.id
+    assert service_db.sms_provider_id == current_sms_provider.id
+
+
+@pytest.mark.parametrize('notification_type', (
+    EMAIL_TYPE,
+    SMS_TYPE
+))
+def test_should_not_update_service_with_inactive_provider(
+    admin_request,
+    notify_db,
```

Comment:
Do we need the notify_db fixture? I believe it will create a db session. Can't we mock it?